flink-table-store-core/src/main/java/org/apache/flink/table/store/table/sink/TableCommit.java

```diff
@@ -18,121 +18,76 @@
+
+    /**
+     * Default ignore empty commit, if this is set to false, when there is no new data, an empty
+     * commit will also be created.
+     *
+     * <p>NOTE: It is recommended to set 'ignoreEmptyCommit' to false in streaming write, in order
+     * to better remove duplicate commits (See {@link #filterCommitted}).
+     */
+    TableCommit ignoreEmptyCommit(boolean ignoreEmptyCommit);
+
+    /** Filter committed commit. This method is used for failover cases. */
+    default boolean filterCommitted(long commitIdentifier) {
+        Set<Long> filter = filterCommitted(Collections.singleton(commitIdentifier));
+        return filter.contains(commitIdentifier);
     }
 
-    public List<ManifestCommittable> filterCommitted(List<ManifestCommittable> committables) {
-        return commit.filterCommitted(committables);
-    }
-
-    public void commit(long identifier, List<FileCommittable> fileCommittables) {
-        ManifestCommittable committable = new ManifestCommittable(identifier);
-        for (FileCommittable fileCommittable : fileCommittables) {
-            committable.addFileCommittable(fileCommittable);
-        }
-        commit(Collections.singletonList(committable));
-    }
-
-    public void commit(List<ManifestCommittable> committables) {
-        if (overwritePartitions == null) {
-            for (ManifestCommittable committable : committables) {
-                commit.commit(committable, new HashMap<>());
-            }
-        } else {
-            ManifestCommittable committable;
-            if (committables.size() > 1) {
-                throw new RuntimeException(
-                        "Multiple committables appear in overwrite mode, this may be a bug, please report it: "
-                                + committables);
-            } else if (committables.size() == 1) {
-                committable = committables.get(0);
-            } else {
-                // create an empty committable
-                // identifier is Long.MAX_VALUE, come from batch job
-                // TODO maybe it can be produced by CommitterOperator
-                committable = new ManifestCommittable(Long.MAX_VALUE);
-            }
-            commit.overwrite(overwritePartitions, committable, new HashMap<>());
-        }
-
-        if (expire != null) {
-            expire.expire();
-        }
-
-        if (partitionExpire != null) {
-            partitionExpire.expire();
-        }
-    }
-
-    @Override
-    public void close() throws Exception {
-        if (lock != null) {
-            lock.close();
-        }
-    }
+    /** Filter committed commits. This method is used for failover cases. */
+    Set<Long> filterCommitted(Set<Long> commitIdentifiers);
+
+    /**
+     * Create a new commit. One commit may generate two snapshots, one for adding new files and the
+     * other for compaction.
+     */
+    void commit(long commitIdentifier, List<CommitMessage> commitMessages);
```

Comment:
Explain what is a `commitIdentifier` and stress that it must become larger for later commits.